### DISCUSSION TITLE
Unable to install the application on ios Simulator

### DISCUSSION THREAD

**Discussion by avcr2019** [airsdk/Adobe-Runtime-Support#2435] [General]
Welcome my friends
I wish you a good day
I would like to try my application on the emulator () but I'm getting this problem.
![Screen Shot 2023-01-18 at 4 26 08 PM](https://user-images.githubusercontent.com/51138931/213198136-165f5ede-3a77-42f7-b37f-25886c23fa97.png)

  Knowing that I put the correct path for the emulator that will run the application.
  
![Screen Shot 2023-01-18 at 4 26 39 PM](https://user-images.githubusercontent.com/51138931/213198164-e76651b6-b7c7-406a-b8f8-7bd7e138d4cd.png)

I hope there is a solution to this problem. Especially since I don't have a real iPhone to test. Is there anything that should be added to the experience of the application on the emulator? Knowing that the emulator is running successfully on program(x code)

**Comment by ajwfrost** [comment]
That error isn't the most helpful one! I wonder what's actually happening.. 
Are you able to set up ADT for some debug logging? Assuming you're using a recent AIR SDK, try creating/editing a file `~/.airsdk/adt.cfg` and add a line `DebugOut=true`. Then do it again, and you should see a log appear in `~/adt.log`.. please let us know what it says there!

thanks

**Reply by dgurney75** [reply]
I've run into this as well and here is the error i see in the log. Not sure where iphone 6 is coming from.

Stack trace:
ADT exception: Invalid simulator option: "iPhone 6" not found
*** ADT Logging at 2023/01/18 15:21:26 ***
Called by: /Users/gurneyd/SDK/AIRSDK_MacOS/lib/adt.jar
ADT 50.1.1.2 called with: -installApp -platform ios -device ios-simulator -package /Users/gurneyd/My Drive/mms_mobile_apps/coreApps/onstarDealerRewards/assets/onStarRewards.ipa -platformsdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk
..java.base/java.lang.Thread.getStackTrace(Thread.java:1606)
..com.adobe.air.DebugFile.printStack(DebugFile.java)
..com.adobe.air.ADTException.<init>(ADTException.java)
..com.adobe.air.ios.IOSSimulatorOperations.mapDeviceType(IOSSimulatorOperations.java)
..com.adobe.air.ios.IOSSimulatorOperations.installApplication(IOSSimulatorOperations.java)
..com.adobe.air.ADT.parsePlatformOperationArgs(ADT.java)
..com.adobe.air.ADT.parseArgsAndGo(ADT.java)
..com.adobe.air.ADT.run(ADT.java)
..com.adobe.air.ADT.main(ADT.java)
Stack trace:
ADT exception: Invalid simulator option: "iPhone 6" not found

**Reply by avcr2019** [reply]
> That error isn't the most helpful one! I wonder what's actually happening.. Are you able to set up ADT for some debug logging? Assuming you're using a recent AIR SDK, try creating/editing a file `~/.airsdk/adt.cfg` and add a line `DebugOut=true`. Then do it again, and you should see a log appear in `~/adt.log`.. please let us know what it says there!
> 
> thanks

Well, my dear friend, I tried and the error appeared already.

`*** ADT Logging at 2023/01/18 23:15:26 ***
Called by: /Users/cetro/Desktop/AIRSDK_MacOS/lib/adt.jar
ADT 50.1.1.2 called with: -version
*** ADT Logging at 2023/01/18 23:16:47 ***
Called by: /Users/cetro/Desktop/AIRSDK_MacOS/lib/adt.jar
ADT 50.1.1.2 called with: -package -target ipa-test-interpreter-simulator -storetype PKCS12 -keystore XYZ -storepass xxxxxx /Users/cetro/Desktop/arabic.ipa /Users/cetro/Desktop/ios Test/ios-app.xml -C /Users/cetro/Desktop/ios Test/ ios.swf Icon-144.png 100.png 50.png Icon-152.png Icon-Small-120.png Icon-Small-80.png Icon-76.png Icon-Small-40.png 180.png Icon-167.png Icon-72.png 48.png 512.png 114.png -C /Users/cetro/Desktop/ios Test/ Assets.car -C /Users/cetro/Desktop/ios Test/ Config.as -extdir /private/var/folders/w2/qtz2sx014wxbshpb0y_s05940000gn/T/TemporaryItems/ios_ANE_Cache -platformsdk Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.2.sdk
*** ADT Logging at 2023/01/18 23:16:47 ***
Called by: /Users/cetro/Desktop/AIRSDK_MacOS/lib/adt.jar
ADT 50.1.1.2 called with: -package -target ipa-test-interpreter-simulator -storetype PKCS12 -keystore XYZ -storepass xxxxxx /Users/cetro/Desktop/arabic.ipa /Users/cetro/Desktop/ios Test/ios-app.xml -C /Users/cetro/Desktop/ios Test/ ios.swf Icon-144.png 100.png 50.png Icon-152.png Icon-Small-120.png Icon-Small-80.png Icon-76.png Icon-Small-40.png 180.png Icon-167.png Icon-72.png 48.png 512.png 114.png -C /Users/cetro/Desktop/ios Test/ Assets.car -C /Users/cetro/Desktop/ios Test/ Config.as -extdir /private/var/folders/w2/qtz2sx014wxbshpb0y_s05940000gn/T/TemporaryItems/ios_ANE_Cache -platformsdk Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.2.sdk
Warning: Using default implementation for native extension 'so.cuo.platform.admob'
compile-abc command line: /Users/cetro/Desktop/AIRSDK_MacOS/lib/aot/bin/compile-abc/compile-abc-64 -mtriple=x86_64-apple-ios11.0 -emit-ane-map

ld64 command line: /usr/bin/ld -pie -ObjC -dead_strip -Z -arch x86_64 -syslibroot "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.2.sdk" -L/usr/lib -F"/System/Library/Frameworks" -L"/Users/cetro/Desktop/AIRSDK_MacOS/lib/aot/lib" -o "../aot9192134936724030837.tmp" -exported_symbol _main -exported_symbol __mh_execute_header -non_global_symbols_no_strip_list "/Users/cetro/Desktop/AIRSDK_MacOS/lib/aot/lib/local_nostrip.x86-simulator-air.txt" "/Users/cetro/Desktop/AIRSDK_MacOS/lib/aot/lib/libRuntimeInterpreter.x86-simulator-air.a" -lc++ -lSystem -U ___darwin_check_fd_set_overflow -lz -lobjc -lclang_rt.ios -framework CoreFoundation -framework UIKit -framework MobileCoreServices -framework CoreGraphics -framework Foundation -framework SystemConfiguration -framework AudioToolbox -framework CFNetwork -framework QuartzCore -framework OpenGLES -framework CoreLocation -framework CoreMedia -framework CoreVideo -framework AVFoundation -framework Security -framework GameController -framework CoreMotion -framework AssetsLibrary -framework WebKit -force_load "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.2.sdk/../../../../../../Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphonesimulator.a" -filelist "/Users/cetro/Desktop/AOTBuildOutput14836996053588473118.tmp/OBJFilesList.txt" -platform_version ios-simulator 11.0 16.0

dsymutil command line: dsymutil -o /Users/cetro/Desktop/arabic.app.dSYM /Users/cetro/Desktop/aot9192134936724030837.tmp

*** ADT Logging at 2023/01/18 23:16:56 ***
Called by: /Users/cetro/Desktop/AIRSDK_MacOS/lib/adt.jar
ADT 50.1.1.2 called with: -uninstallApp -platform ios -device ios-simulator -appid com.arabic -platformsdk Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.2.sdk
..java.base/java.lang.Thread.getStackTrace(Thread.java:1602)
..com.adobe.air.DebugFile.printStack(DebugFile.java)
..com.adobe.air.ADTException.<init>(ADTException.java)
..com.adobe.air.ios.IOSSimulatorOperations.mapDeviceType(IOSSimulatorOperations.java)
..com.adobe.air.ios.IOSSimulatorOperations.uninstallApplication(IOSSimulatorOperations.java)
..com.adobe.air.ADT.parsePlatformOperationArgs(ADT.java)
..com.adobe.air.ADT.parseArgsAndGo(ADT.java)
..com.adobe.air.ADT.run(ADT.java)
..com.adobe.air.ADT.main(ADT.java)
Stack trace:
ADT exception: Invalid simulator option: "iPhone 6" not found
*** ADT Logging at 2023/01/18 23:16:57 ***
Called by: /Users/cetro/Desktop/AIRSDK_MacOS/lib/adt.jar
ADT 50.1.1.2 called with: -installApp -platform ios -device ios-simulator -package /Users/cetro/Desktop/arabic.ipa -platformsdk Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.2.sdk
..java.base/java.lang.Thread.getStackTrace(Thread.java:1602)
..com.adobe.air.DebugFile.printStack(DebugFile.java)
..com.adobe.air.ADTException.<init>(ADTException.java)
..com.adobe.air.ios.IOSSimulatorOperations.mapDeviceType(IOSSimulatorOperations.java)
..com.adobe.air.ios.IOSSimulatorOperations.installApplication(IOSSimulatorOperations.java)
..com.adobe.air.ADT.parsePlatformOperationArgs(ADT.java)
..com.adobe.air.ADT.parseArgsAndGo(ADT.java)
..com.adobe.air.ADT.run(ADT.java)
..com.adobe.air.ADT.main(ADT.java)
Stack trace:
ADT exception: Invalid simulator option: "iPhone 6" not found
`

Here are pictures of the path that I follow and the file that is inside it.
Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.2.sdk
![Screen Shot 2023-01-18 at 11 26 08 PM](https://user-images.githubusercontent.com/51138931/213298448-ae88bb44-0203-4a6f-ace2-ca0df57f9d7f.png)


Here are the types of devices available in my emulator.
![Screen Shot 2023-01-18 at 11 26 58 PM](https://user-images.githubusercontent.com/51138931/213298588-a6575de0-a318-4b00-9c0f-a51a9398bb26.png)

I don't know where the iPhone 6 came from

**Reply by ajwfrost** [reply]
So with both of these, it's calling ADT with `-device ios-simulator`. iPhone 6 is just a default fallback option, you can amend this either by adding to the command line (so `-device ios-simulator -simulator "iPhone 14 Pro Max"` or similar). or you can set an environment variable for the selected simulator into `AIR_IOS_SIMULATOR_DEVICE`. This second option is probably easiest if you're using Animate or another IDE that doesn't let you change the command lines... 

I guess we could try to see whether there's any already-running simulator, and use that as a default, or just the first one it finds, rather than falling back to a named model!

**Reply by avcr2019** [reply]
> So with both of these, it's calling ADT with `-device ios-simulator`. iPhone 6 is just a default fallback option, you can amend this either by adding to the command line (so `-device ios-simulator -simulator "iPhone 14 Pro Max"` or similar). or you can set an environment variable for the selected simulator into `AIR_IOS_SIMULATOR_DEVICE`. This second option is probably easiest if you're using Animate or another IDE that doesn't let you change the command lines...
> 
> I guess we could try to see whether there's any already-running simulator, and use that as a default, or just the first one it finds, rather than falling back to a named model!

Thank you, my friend, for your follow-up and kind reply.
I use Adobe Anime and I wanted to know how to do that step that you mentioned in your previous reply ( you can set an environment variable for the selected simulator into AIR_IOS_SIMULATOR_DEVICE. This second option is probably easiest if you're using Animate or another IDE that doesn't let you change the command lines...). I hope you explain in detail and how to do it because I am excited to try the application on the emulator. As for the first method in which there are codes, if it was working and successfully, a mouse from you or explain how the addition was completed successfully and where this code (So with both of these, it's calling ADT with -device ios-simulator. iPhone 6 is just a default fallback option, you can amend this either by adding to the command line (so -device ios-simulator -simulator "iPhone 14 Pro Max" or similar)) will be added. 
And this is a picture when I go to the experiment
![Screen Shot 2023-01-19 at 2 26 07 AM](https://user-images.githubusercontent.com/51138931/213327404-51f77e80-c93a-4c1c-b3ae-377955029443.png)

Thank you for your help and follow-up, my friend. I wish you a happy day

**Reply by dgurney75** [reply]
Found a way to get the iPhone 6 to work. Just open the iOS Simulator directly.

/Applications/Xcode.app/Contents/Developer/Applications/Simulator.app

Then go to File, New Simulator and enter the name iPhone 6 and select iOS 15.5 and it should work.

**Reply by ajwfrost** [reply]
Cunning workaround! But if it's tricky to set this up, then perhaps we could also provide a configuration item in the adt.cfg file or similar...?

**Reply by avcr2019** [reply]
> Found a way to get the iPhone 6 to work. Just open the iOS Simulator directly.
> 
> /Applications/Xcode.app/Contents/Developer/Applications/Simulator.app
> 
> Then go to File, New Simulator and enter the name iPhone 6 and select iOS 15.5 and it should work.

Thank you, my friend, for sharing with us.
It has already gone to the track as I said as shown in the picture.
![Screen Shot 2023-01-19 at 11 07 43 AM](https://user-images.githubusercontent.com/51138931/213402881-2b7c4d2c-894d-4583-af82-578b7c76f347.png)

But there is a problem. You say we go inside a folder (Simulator.app) and then we follow these steps (Then go to File, New Simulator and enter the name iPhone 6 and select iOS 15.5 and it should work), but there is no option inside this folder to create a new emulator, but the existing one to create a regular folder. Please explain how things happened.
- I tried it, but it was considered 50% successful, which is that I opened the emulator and made a new emulator from the top and choose (new simulator), but it was forcing me to choose its version, which is (16.2).
![Screen Shot 2023-01-19 at 11 21 11 AM](https://user-images.githubusercontent.com/51138931/213403869-b28e9c78-4f85-4391-a324-ca104e6a78a7.png)

 Indeed, the emulator started to work, and I went to (adobe animate) and did the testing, and it was a surprise. The application has been installed successfully and you are logged into the application. But there was a big problem, there is no button that works inside the application. It just opened and only . I think there are certain settings in the build of the iPhone 6.
I hope that you will clarify the last part that you explained in creating the iphone 6 for the test. It may work for me as well as it worked for you. 
I mean this part exactly how I did it (Then go to File, New Simulator and enter the name iPhone 6 and select iOS 15.5 and it should work.)

**Reply by avcr2019** [reply]
> Cunning workaround! But if it's tricky to set this up, then perhaps we could also provide a configuration item in the adt.cfg file or similar...?

I think there is something that needs to be modified or added in order for the emulator to work efficiently. But where I do not know my friend. I just made a new emulator and chose iPhone 6, but made its version 16.2 compulsory. The app is installed but it just opens and I can't navigate inside it. This is a bigger problem than before. Knowing that I sent the application to one of my friends to try it on his iPhone. The application works successfully and without any problems. Is there a solution to run the emulator properly and without problems and complications?
Are we forced to try on iPhone 6? Can we try it on iPhone 7, 8, iPad 10, or any other device we like, or is it compulsory?

**Reply by dgurney75** [reply]
IPhone 6 is just the name. You can select any device using the drop down menu below it. You will need to delete the old one first to change it. Goto devices & simulators in Xcode to do that.

Quick question: Why are you not using the built in mobile Air Emulator inside of Animate?

it would be great to modify this in the config file.

**Reply by avcr2019** [reply]
> IPhone 6 is just the name. You can select any device using the drop down menu below it. You will need to delete the old one first to change it. Goto devices & simulators in Xcode to do that.
> 
> Quick question: Why are you not using the built in mobile Air Emulator inside of Animate?
> 
> it would be great to modify this in the config file.

Thank you for watching, my friend. I wish you a happy day
I do not understand this part yet (IPhone 6 is just the name. You can select any device using the drop down menu below it. You will need to delete the old one first to change it. Goto devices & simulators in Xcode to do tha) or it is not clear to me. I hope you will clarify it in detail.
Regarding your question, yes, I used it for testing and it works successfully, but I did not understand this part (it would be great to modify this in the config file.). I would like to test it on the Apple emulator to see how the application will actually work on a real iPhone. I turned on Icon 6 by chance, as I mentioned in my previous comment, but the application opens and some buttons do not work at all, knowing that on the emulator for Adobe Anime it works successfully, and when exporting to (ipa) it works successfully on my friend's phone. I don't know where is the error? Is it in the emulator? Or in the way I use to run the application on the emulator. It seems that there is some suitable way to run the application successfully and this is what I want to achieve. Especially since I want to try it on the iPad, not just the iPhone. I hope that if the experiment works well for you, explain it in detail so that I can do the experiment because it is possible that the error was on my part in implementing the instructions.

**Reply by dgurney75** [reply]
![Screen Shot 2023-01-19 at 9 38 43 AM](https://user-images.githubusercontent.com/8271407/213486290-28ebbd19-bad5-4c9c-9e48-e5810078a326.png)

**Reply by dgurney75** [reply]
![Screen Shot 2023-01-19 at 9 41 02 AM](https://user-images.githubusercontent.com/8271407/213486546-29d2f5ce-6183-4274-81b2-02bbdfe70f28.png)

**Reply by dgurney75** [reply]
The config comment was meant for Andrew.

**Reply by ajwfrost** [reply]
> it would be great to modify this in the config file.

Easily done .. will be in our next release. Plus a fallback to actually just pick up the first from the list that you get if you run:
```
adt -devices -platform ios -device ios-simulator
```
rather than hard-coding something to "iPhone 6"

**Reply by avcr2019** [reply]
> ![Screen Shot 2023-01-19 at 9 41 02 AM](https://user-images.githubusercontent.com/8271407/213486546-29d2f5ce-6183-4274-81b2-02bbdfe70f28.png)

Please where did you get this list. I can't find it. As for the previous image, I can get it and choose what suits me. Either this image or this interface, where did you get it from, so that I can access it and abide by your instructions and do the experiment.

**Reply by dgurney75** [reply]
You need to open XCode and then select it under Window, Devices and Simulators.

![Screen Shot 2023-01-19 at 12 09 14 PM](https://user-images.githubusercontent.com/8271407/213525904-462f2bb9-0d49-400b-8ae2-16cc6d61f888.png)

**Reply by avcr2019** [reply]
> You need to open XCode and then select it under Window, Devices and Simulators.
> 
> ![Screen Shot 2023-01-19 at 12 09 14 PM](https://user-images.githubusercontent.com/8271407/213525904-462f2bb9-0d49-400b-8ae2-16cc6d61f888.png)

Thank you dear friend . I wish you a happy day .
There is a problem, I do not know whether there is a solution or not. The application opens on the emulator and there is no problem, but all the buttons do not work, while the experiment on the anime works efficiently, and when the experiment is on a real iPhone it works successfully, so why is the problem inside the emulator?

**Reply by dgurney75** [reply]
One thing I noticed on the iOS simulator is that it doesn’t like when Movieclips are on the stage with the alpha set to 0. Make sure they are completely removed or it could affect the behavior of the UI of the other objects below it.

**Reply by avcr2019** [reply]
> One thing I noticed on the iOS simulator is that it doesn’t like when Movieclips are on the stage with the alpha set to 0. Make sure they are completely removed or it could affect the behavior of the UI of the other objects below it.

Thank you for your reply .
Do you mean that I should convert anything from (movieclip) to (button) so that the problem does not occur? Because there is no element in which there is transparency on the stage. But the buttons already were (movieclip). not (button) and I will convert it.
I have a problem. I hope to find a solution for you. I have some files in the format (swf) and call them within the application normally on the Android system, but here in Apple it works as if it is a movie and does not stop, and I do not know what is the reason. Apple company  .

**Reply by dgurney75** [reply]
You can’t load swf files externally with action script on Apple devices. Only works on Android due to Apple’s security requirements. All actionscriot code must be part of the app when installed.

**Reply by avcr2019** [reply]
> You can’t load swf files externally with action script on Apple devices. Only works on Android due to Apple’s security requirements. All actionscriot code must be part of the app when installed.

I know that my friend, but the files are locally embedded. Inside a folder, does this method work, or is it impossible to run the (swf) inside Apple?
Another problem I encountered now is that I put a library of advertisements, and when I press a button, it goes to a specific content, and it succeeds, and when I return, an advertisement appears, and this is very normal, but when I close the advertisement and press any other button on the stage, it does not work. The buttons stop and I don't know why unfortunately. Is it possible to send you the code for verification if you like, or is there a problem in Apple also with the ad libraries?

**Reply by dgurney75** [reply]
See if this example helps. 
[swfLoader.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/10483610/swfLoader.zip)

**Reply by avcr2019** [reply]
> See if this example helps. [swfLoader.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/10483610/swfLoader.zip)

Thank you dear friend
I wish you a happy day
Thank you for the great file, which was useful
But a western problem occurred and I can't find an explanation for it
I took the file as it is and exported it to (ipa) and installed it on the emulator (iphone 6) on the device and it was working great. Press the button and go to view the file (swf) and then press return  and it returns wonderfully and press the button to view the file (launch swf)) once again and works fine without problems.
The problem occurred when I sent the file to my friend, who tried the file on iPhone 7 and iPad 10, and this is what happened in detail: when he pressed the display(launch swf)) button. Already went and display the file (swf) successfully and fine. And when he pressed the return button, he actually returned to the main interface without problems, but the problem that occurred was that when he returned and pressed the display (launch swf)) button again, he did not go to the display again. And here was the surprise, which I could not find an explanation for. I hope you find a solution, my friend. Thank you for following and helping.

edit1 : An explanatory video proving that the file works successfully on the iPhone 6 on the emulator, but on the real iPhone 7 when I return to the main screen and press again. It does not go to the file view.

https://user-images.githubusercontent.com/51138931/214400675-1c84fdf7-f0d5-4272-8221-6277e03b1794.mp4

edit2 : 
Sorry, my friend, there was a problem with iPad 10, which is that the file we are calling is stretched. It does not appear properly. Its size is large and the buttons from above do not appear because of its large size. I put a file (swf) on my own and the same problem. I think this has something to do with the written code.

**Reply by avcr2019** [reply]
> See if this example helps. [swfLoader.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/10483610/swfLoader.zip)

Is there a solution to the problem my dear friend
